src/app/services/courses.service.ts

```diff
@@ -89,12 +87,16 @@ export class CoursesService {
                console.error(error);
                return throwError(error);
             }),
+            tap(() => {
+               this.spinnerService.resolveLoading();
+            }),
             tap(this.saveLastCourseNumber),
             map(this.mapRawCourseItemsToValid)
          );
    }
 
    searchCourses(searchValue: string): Observable<CourseItem[]> {
```

Comment:
Yiu can use `textFragment` as name for parameter and `params: { textFragment }`.